monolith/views/stories.py

```diff
@@ -49,76 +56,73 @@ def _random_story(message=''):
         new_like = Like()
         new_like.liker_id = current_user.id
         new_like.story_id = story_id
         # remove dislike, if present
-        d = Dislike.query.filter_by(disliker_id=current_user.id, story_id=story_id).first()
-        if d is not None: 
+        d = Dislike.query.filter_by(
+            disliker_id=current_user.id, story_id=story_id).first()
+        if d is not None:
             db.session.delete(d)
         db.session.add(new_like)
         db.session.commit()
         message = 'Like added!'
     else:
         message = 'You\'ve already liked this story!'
-    return _story(story_id, message)
-
-@stories.route('/story/<int:story_id>/dislike')
-@login_required
-def _dislike(story_id):
-    story = Story.query.filter_by(id=story_id).first()
-    if story is None:
-        abort(404)
+    return _stories(message)
 
-    q = Dislike.query.filter_by(disliker_id=current_user.id, story_id=story_id)
-    if q.first() is None:
-        new_dislike = Dislike()
-        new_dislike.disliker_id = current_user.id
-        new_dislike.story_id = story_id
-        # remove like, if present
-        l = Like.query.filter_by(liker_id=current_user.id, story_id=story_id).first()
-        if l is not None:
-            db.session.delete(l)
-        db.session.add(new_dislike)
-        db.session.commit()
-        message = 'Dislike added!'
-    else:
-        message = 'You\'ve already disliked this story!'
-    return _story(story_id, message)
 
-@stories.route('/story/<int:story_id>/remove_like')
+@stories.route('/stories/new_story', methods=['GET', 'POST'])
 @login_required
-def _remove_like(story_id):
-    l = Like.query.filter_by(liker_id=current_user.id, story_id=story_id).first()
-    if l is None:
-        message = 'You have to like it first!'
+def new_stories():
+    if request.method == 'GET':
+        dice_themes = retrieve_dice_bytheme()
+        return render_template("new_story.html", themes=dice_themes)
     else:
-        db.session.delete(l)
+        stry = Story.query.filter(Story.author_id == current_user.id).filter(
+            Story.published == 0).filter(Story.theme == request.form["theme"]).first()
+        if stry != None:
+            return redirect("../write_story/"+str(stry.id), code=302)
+
+        dice_set = retrieve_dice_set(request.form["theme"])
+        face_set = dice_set.throw()
+        new_story = Story()
+        new_story.author_id = current_user.id
+        new_story.theme = request.form["theme"]
+        new_story.rolls_outcome = json.dumps(face_set)
+        db.session.add(new_story)
+        db.session.flush()
         db.session.commit()
-        message = 'You removed your like'
-    return _story(story_id, message)
-    
-    
-@stories.route('/story/<int:story_id>/remove_dislike')
+        db.session.refresh(new_story)
+        return redirect('/write_story/'+str(new_story.id), code=302)
+
+
+@stories.route('/write_story/<story_id>', methods=['POST', 'GET'])
```

Comment:
Old code should be used as well. Dislike support is being dropped for some reason.